### PR TITLE
WT-3906 Add config API to connection close for checkpoint.

### PR DESCRIPTION
**Pull request by sueloverso**
Not ready for merge. Add API and new test for forcing the checkpoint on connection close to write to the stable timestamp or to write out all dirty data.

### CODE REVIEW

**Comment by sueloverso** [comment]
@agorrod this is ready for review. I used the same default in `api_data.py` for close as I did for checkpoint.  @dgottlieb FYI.

File: test/suite/test_timestamp12.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            self.session.begin_transaction()
+            c_op[i] = 1
+            c_coll[i] = 1
+            self.pr("i: " + str(i))
+            self.session.commit_transaction(
+              'commit_timestamp=' + timestamp_str(i))
+
+        # Close and reopen the connection.
+        self.close_conn(self.close_cfg)
+        self.conn = self.setUpConnectionOpen(".")
```
Comment (ddanderson) [inline test/suite/test_timestamp12.py]: @sueloverso, I'd change these two lines (starting with `self.conn = `) to just:
```
self.open_conn()
```

File: dist/api_data.py [hunk 2]
```diff
@@ -1295,6 +1295,12 @@ def __cmp__(self, other):
     Config('leak_memory', 'false', r'''
         don't free memory during close''',
         type='boolean'),
+    Config('use_timestamp', 'true', r'''
+        by default, create the close checkpoint as of the last stable timestamp
+        if timestamps are in use, or all current updates if there is no
+        stable timestamp set.  If false, this option generates a checkpoint
+        with all updates including those later than the timestamp''',
```
Comment (bvpvamsikrishna) [inline dist/api_data.py]: In the sentence, "If false", may be better, not to reference timestamp, like "If false, this option generates a checkpoint with all updates"

File: test/suite/test_timestamp12.py (hunk 1 above)
Comment (sueloverso) [reply test/suite/test_timestamp12.py]: Fixed.

File: dist/api_data.py (hunk 2 above)
Comment (sueloverso) [reply dist/api_data.py]: Fixed. Thanks.

**Comment by sueloverso** [comment]
@agorrod this is ready for merge I believe. I addressed the comments from @ddanderson and @bvpvamsikrishna.

**Comment by sueloverso** [comment]
Merging since Alex is on vacation and it has been reviewed by two others.

**Pull request merged**